### DISCUSSION TITLE
Quarkus OpenShift extension - where does trigger version value come from?

### DISCUSSION THREAD

**Discussion by edeandrea** [quarkusio/quarkus#22148] [Q&A]
Using the [Quarkus OpenShift extension](https://quarkus.io/guides/deploying-to-openshift) I'm noticing that in the generated `DeploymentConfig` sometimes the `triggers` section is there and sometimes it isn't.

When its there, it looks like this:

```yaml
triggers:
  - imageChangeParams:
      automatic: true
      containerNames:
        - rest-villains
      from:
        kind: ImageStreamTag
        name: rest-villains:1.0
    type: ImageChange
```

Can someone tell me where that value `1.0` is coming from? Sometimes it shows up as `1.0`, which is the `<version>` in my `pom.xml`, and other times it is coming from the `quarkus.container-image.tag` property.

To generate these files in `target/kubernetes/openshift.yml` I am running this command every time

```base
mvnw clean package -DskipTests \
    -Dquarkus.container-image.tag=java11-latest \
    -Dquarkus.kubernetes.version=java11-latest \
    -Dquarkus.kubernetes.annotations."app.quarkus.io/vcs-url"="$GITHUB_SERVER_URL/$GITHUB_REPOSITORY" \
    -Dquarkus.kubernetes.annotations."app.quarkus.io/vcs-ref"="$GITHUB_REF_NAME" \
    -Dquarkus.openshift.version=java11-latest \
    -Dquarkus.openshift.annotations."app.openshift.io/vcs-url"="$GITHUB_SERVER_URL/$GITHUB_REPOSITORY" \
    -Dquarkus.openshift.annotations."app.openshift.io/vcs-ref"="$GITHUB_REF_NAME"
```

But I am seeing inconsistent results between runs.

You can try it yourself and see.
1. Clone https://github.com/edeandrea/quarkus-super-heroes
2. `cd` into `quarkus-super-heroes`
3. Run `scripts/generate-app-k8s-resources.sh`
4. Look at results in the `rest-villains/deploy/k8s/gen` directory. There will be `app-java11-openshift.yml`,`app-java17-openshift.yml`, `app-native-java11-openshift.yml`, and `app-native-java17-openshift.yml` files and each will be inconsistent as to how the `DeploymentConfig` was generated.

[a comment by an automated account is not shown]

**Comment by edeandrea** [comment]
Anyone?

**Reply by geoand** [reply]
@iocanel do you remember?

**Comment by iocanel** [comment]
The version is the pom version unless the use explicitly specifies a tag value.

AFAIR, for s2i and docker build strategy we generate a trigger. For non open shift container image extensions we don't. Take this with a grain of salt, till I verify.

**Reply by edeandrea** [reply]
I'm seeing some inconsistencies. Sometimes even though I am setting `quarkus.container-image.tag` I end up with the version from the pom.

I'm also seeing that sometimes, even though I set `quarkus.kubernetes.deployment-target=openshift` that sometimes the `DeploymentConfig` contains the trigger and sometimes it doesn't.

In each case the project has all 3 extensions installed (openshift/minikube/kubernetes). I'm setting `quarkus.kubernetes.deployment-target` when I want to generate a specific set of resources.

I'm also setting `quarkus.container-image.builder=docker` in all cases.

**Reply by iocanel** [reply]
> I'm seeing some inconsistencies. Sometimes even though I am setting `quarkus.container-image.tag` I end up with the version from the pom.

This sounds like a bug.

> I'm also seeing that sometimes, even though I set `quarkus.kubernetes.deployment-target=openshift` that sometimes the `DeploymentConfig` contains the trigger and sometimes it doesn't.

 `quarkus.kubernetes.deployment-target=openshift` is what enables generation of `DeploymentConfig` but is not expected to influence the presence of the trigger.

> I'm also setting `quarkus.container-image.builder=docker` in all cases.

The expected behavior in this case is to NOT have a trigger (I verified that). This is a bug and I think that I know the what causes it.

**Reply by edeandrea** [reply]
Ok. Let me know if you need any further info from me. Do you want me to try to put together a reproducer? Want me to create an issue for it?
